### PR TITLE
Support Java using Codaveri & Live Chat for Multi-File Question

### PR DESCRIPTION
**Pull request by bivanalhar**
Note: need to firstly merge https://github.com/Coursemology/coursemology2/pull/7693 to master, since this PR is built on top of that mentioned PR

### CODE REVIEW

File: client/app/bundles/course/assessment/submission/components/answers/Programming/index.jsx [hunk 1]
```diff
@@ -122,29 +121,51 @@ const Programming = (props) => {
...
-          feedbackFiles={feedbackFiles}
-          language={parseLanguages(question.language)}
-          questionId={question.id}
-          readOnly={readOnly}
-          saveAnswerAndUpdateClientVersion={saveAnswerAndUpdateClientVersion}
-        />
-      )}
+    <>
+      <div className="mt-5 flex w-full relative gap-3 mb-1 max-h-[100%]">
+        <Box
```
Comment (cysjonathan) [inline client/app/bundles/course/assessment/submission/components/answers/Programming/index.jsx:126]: nit: if `Box` is not needed then replace with a `div` whenever you come across them ([docs](https://github.com/Coursemology/coursemology2/blob/master/client/README.md#-do-not-use-muis-box-component)).

File: app/services/course/assessment/question/programming_codaveri/java/java_package_service.rb [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  def extract_pathname_from_file(submission_files, pathname)
+    class_name_extractor = /public\s+class\s+([A-Za-z_][A-Za-z0-9_]*)/
+    match = submission_files[pathname].match(class_name_extractor)
+
+    match ? "#{match[1]}.java" : pathname.to_s
+  end
+
+  def strip_autograding_definition_from(file_content)
+    # we strip away all the definitions inside the Autograder class defined within prepend,
+    # which has 6256 characters. Those definitions are not needed to be sent to Codaveri
```
Comment (cysjonathan) [inline app/services/course/assessment/question/programming_codaveri/java/java_package_service.rb:136]: nit: you should also indicate the file you are referencing

File: app/services/course/assessment/programming_codaveri_evaluation_service.rb [hunk 3]
```diff
@@ -137,7 +137,7 @@ def construct_grading_object
     @answer_object[:problemId] = @question.codaveri_id
 
     @answer_object[:languageVersion][:language] = @question.language.polyglot_name
-    @answer_object[:languageVersion][:version] = @question.language.polyglot_version
+    @answer_object[:languageVersion][:version] = @question.language.extend(CodaveriLanguageConcern).polyglot_version
 
     @answer.files.each do |file|
       file_template = default_codaveri_student_file_template
```
Comment (cysjonathan) [inline app/services/course/assessment/programming_codaveri_evaluation_service.rb:143]: if this is a code-submission, don't we also need to extract the filename from the public class?

**Review by cysjonathan — CHANGES_REQUESTED** [review]

File: app/services/course/assessment/programming_codaveri_evaluation_service.rb (hunk 3 above)
Comment (bivanalhar) [reply app/services/course/assessment/programming_codaveri_evaluation_service.rb:143]: done

File: lib/extensions/polyglot_with_database/coursemology/polyglot/language.rb [hunk 4]
```diff
@@ -79,7 +79,7 @@ def polyglot_name
   #
   # @return [String] The language version.
   def polyglot_version
-    name.split[1]
+    (name.split[0].downcase == 'java') ? "#{name.split[1]}.0" : name.split[1]
```
[the remark this answers is not in the document]
Comment (bivanalhar) [reply lib/extensions/polyglot_with_database/coursemology/polyglot/language.rb]: rebased to master already, and with that, the line also got reverted

File: client/app/bundles/course/assessment/submission/components/answers/Programming/index.jsx (hunk 1 above)
Comment (bivanalhar) [reply client/app/bundles/course/assessment/submission/components/answers/Programming/index.jsx]: done

File: app/services/course/assessment/question/programming_codaveri/java/java_package_service.rb (hunk 2 above)
Comment (bivanalhar) [reply app/services/course/assessment/question/programming_codaveri/java/java_package_service.rb]: done

**Pull request merged**